components/pluralrules/src/lib.rs

```diff
@@ -0,0 +1,381 @@
+    ///     .expect("Failed to parse a language identifier.");
+    ///
+    /// let dp = DummyDataProvider::default();
+    ///
+    /// let pr = PluralRules::try_new(lang, PluralRuleType::Cardinal, &dp)
+    ///     .expect("Failed to construct a PluralRules struct.");
+    ///
+    /// match pr.select(1_usize) {
+    ///     PluralCategory::One => "One item",
+    ///     PluralCategory::Other => "Many items",
+    ///     _ => { unreachable!(); }
+    /// };
+    /// ```
+    ///
+    /// The method accepts any input that can be calculated into [`Plural Operands`].
+    /// All unsigned primitive number types can infallibly be converted so they can be
+    /// used as an input.
+    ///
+    /// For signed numbers and strings, [`Plural Operands`] implement `TryFrom` and `FromStr`,
+    /// which should be used before passing the result to `select`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use std::convert::TryFrom;
+    /// # use icu_locale::LanguageIdentifier;
+    /// # use icu_pluralrules::{PluralRules, PluralRuleType};
+    /// use icu_pluralrules::{PluralCategory, PluralOperands};
+    /// # use icu_pluralrules::data::provider::DummyDataProvider;
+    /// #
+    /// # let lang: LanguageIdentifier = "en".parse()
+    /// #     .expect("Failed to parse a language identifier.");
+    /// #
+    /// # let dp = DummyDataProvider::default();
+    /// #
+    /// # let pr = PluralRules::try_new(lang, PluralRuleType::Cardinal, &dp)
+    /// #     .expect("Failed to construct a PluralRules struct.");
+    ///
+    /// let operands = PluralOperands::try_from(-5.2)
+    ///     .expect("Failed to parse to operands.");
+    /// let operands2: PluralOperands = "5.10".parse()
+    ///     .expect("Failed to parse to operands.");
+    ///
+    /// assert_eq!(pr.select(operands), PluralCategory::Other);
+    /// assert_eq!(pr.select(operands2), PluralCategory::Other);
+    /// ```
+    ///
+    /// [`Plural Category`]: ./enum.PluralCategory.html
+    /// [`Plural Operands`]: ./operands/struct.PluralOperands.html
+    pub fn select<I: Into<PluralOperands>>(&self, input: I) -> PluralCategory {
+        self.selector.select(&input.into())
+    }
+
+    /// Returns the LanguageIdentifier associated with this struct.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_locale::LanguageIdentifier;
+    /// use icu_pluralrules::{PluralRules, PluralRuleType, PluralCategory};
+    /// use icu_pluralrules::data::provider::DummyDataProvider;
+    ///
+    /// let lang: LanguageIdentifier = "en".parse()
+    ///     .expect("Failed to parse a language identifier.");
+    ///
+    /// let dp = DummyDataProvider::default();
+    ///
+    /// let pr = PluralRules::try_new(lang, PluralRuleType::Cardinal, &dp)
+    ///     .expect("Failed to construct a PluralRules struct.");
+    ///
+    /// assert_eq!(&pr.locale().to_string(), "en");
+    /// ```
+    pub fn locale(&self) -> &LanguageIdentifier {
```

Comment:
We are discussing in #173 whether to include getters like this.  Since it seems like the one and only purpose of retaining the `locale` field is to power this getter, then maybe remove both the getter and the field until we come to a resolution on #173.